build/pipelines/ci.yml

```diff
@@ -95,3 +95,10 @@ stages:
     jobs:
       - template: ./templates/check-formatting.yml
 
+
+  - stage: CodeIndexer
+    displayName: Github CodeNav Indexer
+    dependsOn: [Build_x64]
+    condition: succeeded()
+    jobs:
+      - template: ./templates/codenav-indexer.yml

```

Comment:
Oof. Doesn't this mean it's going to run on every push and every rolling build? Won't that end up taking a massive amount of storage space for all those indexes to be saved? I would think we would run it only on `main` or something.

Reply:
That is a good question. Lemme bring in the codenav folks to see what they think

Reply:
Hi! Holly from the codenav team here! We're fine with this running with such high frequency and the storage load, we actually want to test how well we scale to highly active repos. I will say, most folks who onboard do only run our task on builds of main and a few teams have made a separate .yml file just for the build+codenav tasks so that they can have more fine-tuned control of when it runs, but it's up to you what builds you would find benefit from having navigation on.

Reply:
My bigger concern is that this will build against 3p code on _pull requests_. We might want to restrict it like we did the ARM64 build :smile: